From 11dbb761ff385095051b5e4f1df4d69eb283a7d0 Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Fri, 10 Feb 2017 17:34:03 -0500
Subject: [PATCH 03/20] hsr: introduce rx offload feature flags for HSR and PRP

This patch introduces the HSR/PRP rx offload flag to allow offload
of LRE rx side processing such as duplicate detect/drop, node_table,
etc to h/w. Lower level Ethernet drivers that support h/w with these
protocol handling list this feature in the feature flag of the
net_device.

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
---
 include/linux/netdev_features.h | 4 ++++
 net/core/ethtool.c              | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/include/linux/netdev_features.h b/include/linux/netdev_features.h
index f0d8734..f0a7fb4 100644
--- a/include/linux/netdev_features.h
+++ b/include/linux/netdev_features.h
@@ -66,6 +66,8 @@ enum {
 	NETIF_F_HW_VLAN_STAG_FILTER_BIT,/* Receive filtering on VLAN STAGs */
 	NETIF_F_HW_L2FW_DOFFLOAD_BIT,	/* Allow L2 Forwarding in Hardware */
 	NETIF_F_BUSY_POLL_BIT,		/* Busy poll */
+	NETIF_F_HW_HSR_RX_OFFLOAD_BIT,	/* HSR Rx handling in hardware/firmware */
+	NETIF_F_HW_PRP_RX_OFFLOAD_BIT,	/* PRP Rx handling in hardware/firmware */
 
 	/*
 	 * Add your fresh new feature above and remember to update
@@ -124,6 +126,8 @@ enum {
 #define NETIF_F_HW_VLAN_STAG_TX	__NETIF_F(HW_VLAN_STAG_TX)
 #define NETIF_F_HW_L2FW_DOFFLOAD	__NETIF_F(HW_L2FW_DOFFLOAD)
 #define NETIF_F_BUSY_POLL	__NETIF_F(BUSY_POLL)
+#define NETIF_F_HW_HSR_RX_OFFLOAD __NETIF_F(HW_HSR_RX_OFFLOAD)
+#define NETIF_F_HW_PRP_RX_OFFLOAD __NETIF_F(HW_PRP_RX_OFFLOAD)
 
 #define for_each_netdev_feature(mask_addr, bit)	\
 	for_each_set_bit(bit, (unsigned long *)mask_addr, NETDEV_FEATURE_COUNT)
diff --git a/net/core/ethtool.c b/net/core/ethtool.c
index 29edf74..06aa21f 100644
--- a/net/core/ethtool.c
+++ b/net/core/ethtool.c
@@ -98,6 +98,8 @@ static const char netdev_features_strings[NETDEV_FEATURE_COUNT][ETH_GSTRING_LEN]
 	[NETIF_F_RXALL_BIT] =            "rx-all",
 	[NETIF_F_HW_L2FW_DOFFLOAD_BIT] = "l2-fwd-offload",
 	[NETIF_F_BUSY_POLL_BIT] =        "busy-poll",
+	[NETIF_F_HW_HSR_RX_OFFLOAD_BIT] = "hsr-rx-offload",
+	[NETIF_F_HW_PRP_RX_OFFLOAD_BIT] = "prp-rx-offload",
 };
 
 static const char
-- 
1.9.1

